spk/pyload/src/installer.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/sh
+
+# Package
+PACKAGE="pyload"
+DNAME="pyLoad"
+
+# Others
+INSTALL_DIR="/usr/local/${PACKAGE}"
+SSS="/var/packages/${PACKAGE}/scripts/start-stop-status"
+PYTHON_DIR="/usr/local/python"
+PATH="${INSTALL_DIR}/bin:${INSTALL_DIR}/env/bin:${PYTHON_DIR}/bin:${PATH}"
+USER="pyload"
+GROUP="users"
+VIRTUALENV="${PYTHON_DIR}/bin/virtualenv"
+CFG_FILE="${INSTALL_DIR}/etc/pyload.conf"
+TMP_DIR="${SYNOPKG_PKGDEST}/../../@tmp"
+
+SERVICETOOL="/usr/syno/bin/servicetool"
+FWPORTS="/var/packages/${PACKAGE}/scripts/${PACKAGE}.sc"
+
+lng2iso()
+{
+	# changes 3-character Synology language code to ISO 639-1 code.
+        case $1 in
+            eng|deu|fre|ita|nld|sve|rus|plk|csy)
+                echo "${1%?}"
+                ;;
+            spn)
+                echo "es"
+                ;;
+            *)
+		echo "en"
+		;;
+        esac
+}
+
+
+preinst ()
+{
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        # Check directory
+        if [ ! -d ${wizard_download_dir:=/volume1/downloads} ]; then
+            echo "Download directory ${wizard_download_dir:=/volume1/downloads} does not exist."
+            exit 1
+        fi
+    fi
+    exit 0
+}
+
+postinst ()
+{
+    # Link
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+    # Create a Python virtualenv
+    ${VIRTUALENV} --system-site-packages ${INSTALL_DIR}/env > /dev/null
+
+    # Create user
+    adduser -h ${INSTALL_DIR}/etc -g "${DNAME} User" -G ${GROUP} -s /bin/sh -S -D ${USER}
+
+    # Set cfg location
+    echo "${INSTALL_DIR}/etc" > "${INSTALL_DIR}/share/pyload/module/config/configdir"
+
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        # Set group and permissions on download dir for DSM5
+        if [ `/bin/get_key_value /etc.defaults/VERSION buildnumber` -ge "4418" ]; then
+            chgrp users "${wizard_download_dir:=/volume1/downloads}"
+            chmod g+rwx "${wizard_download_dir:=/volume1/downloads}"
+        fi
+
+        # Edit the configuration according to the wizard
+        sed -i -e "s|@W_DOWNLOAD_DIR@|${wizard_download_dir:=/volume1/downloads}|" \
+               -e "s|@LNG@|$(lng2iso ${SYNOPKG_DSM_LANGUAGE})|" \
```

Comment:
I'm wondering if it's worth changing the interface language based on the device language, instead of just defaulting to 'en'? I assume you can change the language via the interface, right?